apps/kyb-app/src/components/organisms/UIRenderer/elements/JSONForm/components/JSONFormArrayFieldLayout/JSONFormArrayFieldLayout.tsx

```diff
@@ -55,14 +58,15 @@ export const JSONFormArrayFieldLayout = (props: ArrayFieldsLayoutProps) => {
   // To make validation properly handle addition of items we have to push empty object to array at destination.
   const addEmptyElementToDestinationArray = useCallback(() => {
     let value: undefined | any[] = get(payload, definition.valueDestination as string);
+    const { defaultValue = {} } = definition?.options || {};
 
     if (Array.isArray(value)) {
-      value.push({});
+      value.push(defaultValue);
     } else {
-      value = [{}];
+      value = [defaultValue];
     }
 
-    set(payload, definition.valueDestination as string, value);
+    set(payload, definition.valueDestination as string, [...value]);
```

Comment:
`set` mutates `payload` are you sure it won't cause bugs in the context of React?

Reply:
`setContext` called right after which triggers rerender. Thats fine.

Reply:
Any reason we're not using some `setState` over `set`?

Reply:
Its not component state, its the whole context. Path to value is dynamic so `set` to assign value at destination. Then state is updated so app will know that `context` updated.